### DISCUSSION TITLE
SPFx 1.14 BETA - Feedback Collected

### DISCUSSION THREAD

**Discussion by StfBauer** [SharePoint/sp-dev-docs#7568] [Show and tell]
# Confusing template / Framework selection

<img width="451" alt="Screenshot 2021-12-08 at 23 07 50" src="https://user-images.githubusercontent.com/5503835/145292390-fc15a5a5-8790-4a60-ab54-c108c8f29821.png">

From a generator developer experience this makes sense - but not from a developer experience.

- No Framework 
- React

Are two variants of the web part.

Minimal would assume this is another Framework!?! A better option would be to ask for Sample content via Yes / No afterwards.
Also, what should I choose if I like to have Minimal and HTML or Minimal and ReactJS?

Is Minimal using ReactJS on NoFramework.

**Comment by StfBauer** [comment]
# Deprecated extension

Please remove the deprecated Chrome Extension configuration from the generated project. It is deprecated.

<img width="483" alt="image" src="https://user-images.githubusercontent.com/5503835/145293420-fe0c0180-0175-468b-b239-06a81694fcf7.png">

**Reply by andrewconnell** [reply]
Agreed... see #7558

**Comment by StfBauer** [comment]
# SASS upgrade needed

SASS issue in web part currently it still used: [@import from SASS](https://sass-lang.com/documentation/at-rules/import) should not be used anymore looking in the future.

```scss
@import '~@microsoft/sp-office-ui-fabric-core/dist/sass/SPFabricCore.scss';
```

What is should be for the future the link should go like this:

```scss
@use '~@microsoft/sp-office-ui-fabric-core/dist/sass/SPFabricCore.scss';
```

With use it creates the following error message: 
<img width="684" alt="image" src="https://user-images.githubusercontent.com/5503835/145299340-84c71f87-58ab-4f50-8017-0d7ff5404ef0.png">

The error message is because the package is not made to be used in newer versions of SASS. It is also based on the very very old Office UI Fabric Core implementation.

In order to make this work in future Offfice UI Fabric Core needs updated:
<img width="321" alt="image" src="https://user-images.githubusercontent.com/5503835/145301288-fc59fb45-6b84-45c9-a228-7c7fa3609049.png">

I guess this is unlikely so it means that also `sp-office-ui-fabric-core` is dead and needs an upgrade.

<img width="397" alt="image" src="https://user-images.githubusercontent.com/5503835/145302080-c02f7274-77e5-447d-bb4b-666fec4f0e23.png">

**Comment by StfBauer** [comment]
# There is no Dark Theme in SharePoint or Fluent UI

There is technically no dark theme in SharePoint just an inverted one.

<img width="625" alt="image" src="https://user-images.githubusercontent.com/5503835/145302574-d84c1c07-016c-4ad1-b5e2-557f991f6e58.png">

This is a defined dark theme in the theme generator, which also means that it is not inverted but defined as a dark theme.

<img width="1547" alt="image" src="https://user-images.githubusercontent.com/5503835/145302844-55e62395-6a32-4617-8494-4330fafce2c7.png">

The only thing that matters for inverted is that the primary colour and background colour will be swapped. 

I can define a light theme and make it dark by setting it isInverted to true - Which gives a dark theme
In the same way, I can define a dark theme and setting it to isInverted to true - Which gives a light theme

isInverted is correct - isDarkTheme is wrong.

**Comment by StfBauer** [comment]
# Semantic color slots only:

<img width="725" alt="image" src="https://user-images.githubusercontent.com/5503835/145304133-3d5b28bc-5509-4972-98fb-55a2d977979d.png">

There are more useful than semantic color slots because these might change in future and there is also the possibility that the meaning and naming of these slots change. A safer approach is to use color palettes. These are more reliant even for updates in the future.

These remain stable:
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/5503835/145304391-b917be71-0e87-40b9-b6e0-074e045993fe.png">

These might change in future:
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/5503835/145304450-9b826b56-9f64-4a78-a9ad-910a610da886.png">

For the future, it would also be good to have a comment in there like for 'all semantic colour slots look at the theme generator'.

## Other import color slots:
<img width="507" alt="image" src="https://user-images.githubusercontent.com/5503835/145306286-ffb6f582-a3ec-4f0f-be00-e7e9b22970ba.png">

**Comment by StfBauer** [comment]
# Why selective CSS Variable and only onThemeChanged?

No that the theme support got abstracted away, I do not understand why onThemeChanged is still there? It is good to maybe react to theme changes but why do I have to reset selectively some CSS Variables.

<img width="777" alt="image" src="https://user-images.githubusercontent.com/5503835/145305496-b7a6358a-7c3c-4063-9edb-8d548538195c.png">

## Developer assumption

SPFx change changed the give CSS Variables on their own and in 90% of the cases, I don't have to do anything. Setting just partially some variable  Do not make sense in this case.

What happens if the theme changes on a Full Canvas Page? 
Is this already fixed and returns the proper theme now?
As I checked this onThemeChanged sets the CSS Variables on the web part event without changing the theme.

<img width="688" alt="image" src="https://user-images.githubusercontent.com/5503835/145306528-1eb09c59-9d7e-4e58-86b6-6512498a560b.png">

## 🧞‍♀️Wish 

There should be the complete Office UI Fabric theme slots available, not just a view, Register all CSS properties is no performance impact and also do not matter if they are registered before the web part is rendered or after.

CSS Variables will be automatically applied to the code inside the web part.

In future, the CSS Variables should be available in the section of the page or on the body. Since CSS Variables are scoped to the HTML Element only they can be flexible overwritten.

Happy to help on that. ;)

**Comment by andrewconnell** [comment]
> Can't figure out how to reply to the [initial feedback for the template picker above...](https://github.com/SharePoint/sp-dev-docs/discussions/7568#discussion-3731027)

# Confusing template / Framework selection

100% agree with @StfBauer ... the three selection options are confusing. Seems there are 4 possible permutations of project templates:

1. no JS - no theme & no default content
1. no JS - default theme & default content
1. React - no theme & no default content
1. React - default theme & default content

But from the above list, the provided options don't support option 3 above.

My recommendation... change what's currently presented:

- Minimal 👉  _{no change}_
  - *minimal means I don't want anything... no theme, no web framework, etc... just a working component*
- No JavaScript framework 👉  No web framework (with theme support)
- ReactJS 👉  ReactJS (with theme support)

Ideally, there's one more template added:
- ReactJS 👉  ReactJS (without theme support)

Furthermore, the above reflects the following: **we should move away from "JavaScript Framework" in the terminology**. That's not what we're talking about ([proof](https://www.google.com/search?q=is+react+a+javascript+framework)). React != JavaScript framework... it's a _JavaScript library for building UI_ OR it's a _web framework_... it's not a _JavaScript Framework_. If we're changing things, then why not take advantage of the opportunity to get it right this time.

**Reply by sparkitect** [reply]
I'd be ok with two separate questions:
1. Web Framework: None | ReactJS
2. Theme support? Yes | No
